public/app/features/search/components/DashboardActions.tsx

```diff
@@ -18,7 +18,7 @@ export const DashboardActions: FC<Props> = ({ folderUid, canCreateFolders = fals
       url = `dashboards/folder/new/`;
     }
 
-    if ((isTypeNewFolder && config.featureToggles.nestedFolders) || folderUid) {
+    if ((isTypeNewFolder && config.featureToggles.nestedFolders && folderUid) || folderUid) {
```

Comment:
Isn't this change enough? Then we never get into the scenario where the `queryParams.folderUid` is `undefined`? Also I believe this can just be `if(folderUid)` now because if `folderUid` is truthy this will always be `true` because of what is after the `||`.

Reply:
you are right! thanks Joao